components/raftstore/src/store/fsm/apply.rs

```diff
@@ -1135,7 +1137,11 @@ where
                 has_unflushed_data = false;
             }
             if self.priority != apply_ctx.priority {
-                if has_unflushed_data {
+                if has_unflushed_data
+                    && apply_ctx
```

Comment:
When will it return false?

Reply:
Always in TiFlash

Reply:
TiKV share writebatch across peers. Not flushing the writebatch can reorder the apply. For example if a peer put(k1, v1) to writebatch 1, and put(k1, v2) to writebatch 2. The final value of k1 depends on which writebatch is written first.

Reply:
In TiFlash‘s scenario, we guarantee there is no real KV data in WriteBatch. We hereby prevent a persist here to persist advanced apply state.